v23.1/architecture/distribution-layer.md

```diff
@@ -87,7 +87,9 @@ In relationship to other layers in CockroachDB, the distribution layer:
 
 gRPC is the software nodes use to communicate with one another. Because the distribution layer is the first layer to communicate with other nodes, CockroachDB implements gRPC here.
 
-gRPC requires inputs and outputs to be formatted as protocol buffers (protobufs). To leverage gRPC, CockroachDB implements a protocol-buffer-based API defined in `api.proto`.
+gRPC requires inputs and outputs to be formatted as protocol buffers (protobufs). To leverage gRPC, CockroachDB implements a protocol-buffer-based API.
+
+By default, the network timeout for RPC connections between nodes is {{site.data.cockroach_network_timeout}}, with a connection timeout of {{site.data.cockroach_network_connection_timeout}}, in order to reduce unavailability and tail latencies during infrastructure outages. This can be changed via the environment variable `COCKROACH_NETWORK_TIMEOUT` which defaults to {{site.data.cockroach_network_timeout}}.
```

Comment:
This is only kind of true. gRPC network timeouts only apply from the server's point of view, not the client's (in this case the client is another CRDB node), and the client is the one that usually matters. We additionally send an RPC ping every second to check that RPC connections are alive, but these have a fairly high timeout of 6 seconds (3x network timeout) because they're subject to contention with RPC traffic (so e.g. during a large import they can get delayed by several seconds).

This additional complexity is unfortunate, but currently necessary -- ideally we would only rely on the gRPC network timeout, but gRPC doesn't let us do that from the client side.

Reply:
I added another paragraph with this info, slightly revised. PTAL since I may have misunderstood a bit

Reply:
Yep, that's accurate, thanks!